src/document/DocumentAnnotator.ts

```diff
@@ -12,10 +13,31 @@ export default class DocumentAnnotator extends BaseAnnotator {
 
     managers: Map<number, Set<BaseManager>> = new Map();
 
+    selectionChangeTimer: NodeJS.Timer | null = null;
+
     constructor(options: Options) {
         super(options);
 
         this.addListener(Event.ANNOTATIONS_MODE_CHANGE, this.handleChangeMode);
+        document.addEventListener('selectionchange', this.handleSelectionChange);
+    }
+
+    destroy(): void {
+        if (this.selectionChangeTimer) {
+            clearTimeout(this.selectionChangeTimer);
+        }
+
+        this.removeListener(Event.ANNOTATIONS_MODE_CHANGE, this.handleChangeMode);
+        document.removeEventListener('selectionchange', this.handleSelectionChange);
+
+        super.destroy();
```

Comment:
Should this be done first?

Reply:
I remember in other language, parent's destructor must be put in the end of child's destructor. Just want to follow the same convention in JS.

Reply:
lol that makes sense